### PR TITLE
[Doc] Add pcolor, contour, imshow to  and other small changes

### PR DESCRIPTION
**Pull request by jklymak**
## PR Summary

  - Added `imshow`, `pcolor`, and `contour` to `../tutorials/introductory/sample_plots.py`
  - Changed `command` to `function`
  - Removed `Demo` from many of the titles.  

Redo of #9387 (sorry @anntzer, who already commented)

## PR Checklist

- [x] Documentation is sphinx and numpydoc compliant

### CODE REVIEW

File: tutorials/introductory/sample_plots.py [hunk 1]
```diff
@@ -54,7 +88,7 @@
 
 .. _screenshots_path_demo:
 
-Path demo
+Paths
 =========
```
Comment (afvincent) [inline tutorials/introductory/sample_plots.py]: I agree that the current version is valid RST, but would it be possible to match the length of the '====' line with the line beyond?

File: tutorials/introductory/sample_plots.py [hunk 2]
```diff
@@ -168,10 +202,10 @@
 
 .. _screenshots_table_demo:
 
-Table demo
+Tables
 ==========
```
Comment (afvincent) [inline tutorials/introductory/sample_plots.py]: Same as before, but would it be possible to match the length of the '====' line with the line beyond?

File: tutorials/introductory/sample_plots.py [hunk 3]
```diff
@@ -349,7 +383,7 @@
 
 .. _screenshots_eeg_demo:
 
-EEG demo
+EEG GUI
 =========
```
Comment (afvincent) [inline tutorials/introductory/sample_plots.py]: Same as before: would it be possible to match the length of the '====' line with the line beyond?

File: tutorials/introductory/sample_plots.py [hunk 4]
```diff
@@ -36,12 +36,46 @@
 
    Subplot
 
+.. _screenshots_images_demo:
+
+Images
+======
+
+Matplotlib can display images (assuming equal-spaced, equal-aspect-ratio
+horizontal dimensions) using the :func:`~matplotlib.pyplot.imshow` function.
```
Comment (anntzer) [inline tutorials/introductory/sample_plots.py]: 1. you don't actually need equal aspect ratio
2. something seems to be missing before or after "horizontal dimensions"?

**Comment by NelleV** [comment]
Thanks a lot @jklymak !

**Comment by tacaswell** [comment]
@meeseeksdev backport to v2.1.0-doc

**Pull request merged**